usl_models/usl_models/atmo_ml/data_utils.py

```diff
@@ -51,10 +54,19 @@ def split_time_step_pairs(data: tf.Tensor) -> tf.Tensor:
     Returns:
         A time sequence of single tensors of shape [B, 2T, H, W, C].
     """
-    # We need to permute the tensor to place the time and channel axes together
-    # during the reshape, in order to avoid disturbing the spatial dimensions.
+    # Transpose to [B, H, W, T, 2C] so that we can split the channel dimension
     permuted = tf.transpose(data, (0, 2, 3, 1, 4))
-    shape = permuted.shape[:3] + (permuted.shape[-2] * 2, permuted.shape[-1] // 2)
-    reshaped = tf.reshape(permuted, shape)
+
+    # Get the dynamic shape
+    batch_size = tf.shape(permuted)[0]
```

Comment:
I generally prefer unpacking these in one line, like:
```
B, H, W, T_HALF, C_PAIR= tf.shape(permuted)
timesteps = 2 * T
channels = C // 2
```

Reply:
Done!

Reply:
when doing this, i got an error in check:
"File "/home/runner/work/climateiq-cnn/climateiq-cnn/usl_models/usl_models/atmo_ml/data_utils.py", line 61, in split_time_step_pairs  *
            B, H, W, T_HALF, C_PAIR = tf.shape(permuted)
    
        OperatorNotAllowedInGraphError: Iterating over a symbolic `tf.Tensor` is not allowed. You can attempt the following resolutions to the problem: If you are running in Graph mode, use Eager execution mode or decorate this function with @tf.function. If you are using AutoGraph, you can try decorating this function with @tf.function. If that does not work, then you may be using an unsupported feature or your source code may not be visible to AutoGraph."